java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/UserdataService.java

```diff
@@ -275,10 +328,12 @@ public ResponseEntity<String> webUserRegister(@RequestParam(name = "email", requ
 		if (pu == null) {
 			throw new OsmAndPublicApiException(ERROR_CODE_EMAIL_IS_INVALID, "email is not registered");
 		}
-		String errorMsg = userSubService.checkOrderIdPremium(pu.orderid);
-		if (errorMsg != null) {
-			throw new OsmAndPublicApiException(ERROR_CODE_NO_VALID_SUBSCRIPTION, errorMsg);
-		}
+        if (pu.orderid != null) {
+            String errorMsg = userSubService.checkOrderIdPremium(pu.orderid);
```

Comment:
This is strange check cause we should allow register without valid subscription it's not clear why do we check premium subscription